internal/controller/shared.go

```diff
@@ -203,13 +203,30 @@ func reconcileStateStoreCommon[T metav1.Object](
 	resourceType string,
 	updateStatus func(stateStore T, failureReason string, failureMessage string, err error) error,
 ) (ctrl.Result, error) {
-	secret := &corev1.Secret{}
+	if secretRef == nil {
+		err := fmt.Errorf("secretRef is empty")
+		if err := updateStatus(stateStore, StateStoreNotReadySecretErrorReason, StateStoreNotReadySecretErrorMessage, err); err != nil {
+			o.logger.Error(err, "error updating state store status")
+		}
+		return ctrl.Result{}, err
```

Comment:
```suggestion
		return ctrl.Result{}, nil
```

there's no point in returning the error here; returning an error tells kubernetes to try the reconcilation again, but this will only succeed when the object is updated -- and that will trigger a reconciliation itself.

Reply:
actually this should not even error 🤔

There are two scenarios:

* secretRef is required

If this field is required, we should make it required in the  CRD, making this entire block overly defensive -- i'd just remove make that change and remove this check (kubernetes should enforce it for us).

* secretRef is optional

If its optional, it should error; we should not check anything and move on

---

We already have a scenario where this is optional: when we are using IAM authMethod (see the [bucket state store documentation](https://docs.kratix.io/main/reference/statestore/bucketstatestore)). We should only run the ""does the secret exist"  check when the secretRef has been provided. If not, we should just skip this validation altogether.

Reply:
Didn't know about the case where it's not required, thanks 👍

On further review I can see that we already check the `secretRef` properly in `newWriter`, which errors in the appropriate scenarios you mention, so I'll just defer to that function and remove the secret checks entirely!

Reply:
Fixed in https://github.com/syntasso/kratix/pull/400/commits/f04ef84dea11035518d3d751b110380402811668 - good spot, thanks!